components/brave_wallet/common/mem_utils.h

```diff
@@ -31,7 +33,8 @@ struct SecureZeroAllocator {
     return static_cast<T*>(::operator new(n * sizeof(T)));
   }
   void deallocate(T* p, size_t n) {
-    SecureZeroData(p, n);
+    SecureZeroData(
+        base::as_writable_bytes(UNSAFE_BUFFERS(base::make_span(p, n))));
```

Comment:
Did we have a potential bug there? It was allocating n * sizeof(T) bytes, but clearing only first n bytes. Works correctly only when sizeof(T) == 1 which seems the only way it was instantiated with.

Reply:
There was indeed a bug here that was flying undetected. There was also some very bad UB in the tests for the `SecureZeroData` function.